src/platform/scene/spawn-point.ts

```diff
@@ -0,0 +1,75 @@
+import { generateValidator, JSONSchema, ValidateFunction } from "../../validation";
+
+/** @alpha */
+export type SpawnPoint = {
+  name?: string;
+  position: {
+    x: number | number[];
+    y: number | number[];
+    z: number | number[];
+  };
+  default?: boolean;
+  cameraTarget?: {
+    x: number;
+    y: number;
+    z: number;
+  };
+};
+
+/** @alpha */
+export namespace SpawnPoint {
+  export const schema: JSONSchema<SpawnPoint> = {
+    type: "object",
+    properties: {
+      name: {
+        type: "string",
+        nullable: true,
+      },
+      position: {
+        type: "object",
+        properties: {
+          x: { $ref: "#position" },
+          y: { $ref: "#position" },
+          z: { $ref: "#position" },
+        },
+        additionalProperties: false,
+        required: ['x', 'y', 'z']
+      },
+      default: {
+        type: "boolean",
+        nullable: true
+      },
+      cameraTarget: {
+        type: 'object',
+        properties: {
+          x: { type: "number" },
+          y: { type: "number" },
+          z: { type: "number" },
+        },
+        additionalProperties: false,
+        required: ['x', 'y', 'z'],
+        nullable: true
+      }
+    },
+    definitions: {
+      // @ts-ignore
```

Comment:
Needed to add this, because typings didn't work correctly with `number | number[];`

Reply:
I left a comment with a suggestion